llama_pipe.py

```diff
@@ -221,7 +221,13 @@ def initial_layer(inputs):
         for block in self.model.layers:
             result.append(LayerSpec(LlamaDecoderLayerPipe, self.loader_util, block))
         result.append(LayerSpec(LlamaRMSNormPipe, self.loader_util, self.model.norm, _estimated_size=0))
-        result.append(LayerSpec(LmHeadPipe, self.loader_util, self.lm_head, _estimated_size=0))
+        result.append(LayerSpec(
+            LmHeadPipe,
+            self.loader_util,
+            self.lm_head,
+            tie_weights='model.embed_tokens.weight' if self.config.get('tie_word_embeddings', False) else None,
```

Comment:
```suggestion
            tie_weights='model.embed_tokens.weight' if self.config.tie_word_embeddings else None,
```

Reply:
How will python handle this if there is no `tie_word_embeddings` present, say for the older llama or qwen models? That was the only reason I did it like this:

> `if self.config.get('tie_word_embeddings', False)`

returns True iff it exists and is set to True.

> `self.config.tie_word_embeddings`

I would assume that this will throw a runtime exception if it is missing, but I don't really know the ins and outs of python very well?

Reply:
`config` is a `LoraConfig` and will always have a `tie_word_embeddings`, I believe. It's based on the version of PEFT and doesn't rely on the underlying model.

Reply:
Confirmed the above in testing. See comment below.